xtable-core/src/main/java/org/apache/xtable/parquet/ParquetPartitionHelper.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.xtable.parquet;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.xtable.model.schema.InternalPartitionField;
+import org.apache.xtable.model.schema.InternalSchema;
+import org.apache.xtable.model.schema.PartitionTransformType;
+import org.apache.xtable.model.stat.PartitionValue;
+import org.apache.xtable.model.stat.Range;
+import org.apache.xtable.schema.SchemaFieldFinder;
+
+public class ParquetPartitionHelper {
```

Comment:
nitpick on naming, "helper" is a bit vague and ends up being a place where developers will dump all sorts of functionality. This is a `ParquetPartitionExtractor` based on the naming conventions we've used in other formats